pkg/agent/controller/networkpolicy/allocator.go

```diff
@@ -64,13 +64,18 @@ func asyncRuleCacheKeyFunc(obj interface{}) (string, error) {
 
 // newIDAllocator returns a new *idAllocator.
 // It takes a list of allocated IDs, which can be used for the restart case.
-func newIDAllocator(allocatedIDs ...uint32) *idAllocator {
+func newIDAllocator(flowPollInterval time.Duration, allocatedIDs ...uint32) *idAllocator {
 	allocator := &idAllocator{
 		availableSet:   make(map[uint32]struct{}),
 		asyncRuleCache: cache.NewStore(asyncRuleCacheKeyFunc),
 		deleteQueue:    workqueue.NewNamedDelayingQueue("async_delete_networkpolicyrule"),
 	}
 
+	// Set the asyncDeleteInterval based on flowPollInterval value.
+	if asyncDeleteInterval < flowPollInterval {
+		asyncDeleteInterval = flowPollInterval
+	}
+
```

Comment:
I feel like `flowPollInterval` does not really belong in this code. Maybe the parameter should be named `asyncDeleteInterval` instead and the current `asyncDeleteInterval` variable should be renamed to `minAsyncDeleteInterval`

Reply:
Yes, I was not comfortable too. This is a good suggestion. Changed other parts as well to keep flow poll interval separate.